### PR TITLE
Give driver a  map of handler names to client.TargetHandler

### PR DESCRIPTION
**Pull request by davis-haba**
Signed-off-by: davis-haba <davishaba@google.com>

### CODE REVIEW

**Comment by davis-haba** [comment]
@willbeason

File: constraint/pkg/client/drivers/local/local.go [hunk 1]
```diff
@@ -78,6 +79,7 @@ type Driver struct {
 	printHook     print.Hook
 	providerCache *externaldata.ProviderCache
 	externs       []string
+	handlers      map[string]client.TargetHandler
```
Comment (willbeason) [inline constraint/pkg/client/drivers/local/local.go:82]: Add a comment that this is a map from handler name to the respective handler.

**Review by willbeason — CHANGES_REQUESTED** [review]
Two minor changes, otherwise looks good

File: constraint/pkg/client/drivers/local/args.go [hunk 2]
```diff
@@ -98,3 +99,19 @@ func DisableBuiltins(builtins ...string) Arg {
 		d.capabilities.Builtins = nb
 	}
 }
+
+<<<<<<< HEAD
```
Comment (maxsmythe) [inline constraint/pkg/client/drivers/local/args.go:103]: Merge conflict indicators

File: constraint/pkg/client/drivers/local/args.go [hunk 3]
```diff
@@ -98,3 +99,11 @@ func DisableBuiltins(builtins ...string) Arg {
 		d.capabilities.Builtins = nb
 	}
 }
+
+func Handlers(handlers ...client.TargetHandler) Arg {
+	return func(d *Driver) {
```
[the remark this answers is not in the document]
Comment (davis-haba) [reply constraint/pkg/client/drivers/local/args.go:104]: done

File: constraint/pkg/client/drivers/local/local.go (hunk 1 above)
Comment (davis-haba) [reply constraint/pkg/client/drivers/local/local.go]: done

File: constraint/pkg/client/drivers/local/args.go (hunk 2 above)
Comment (davis-haba) [reply constraint/pkg/client/drivers/local/args.go]: Yep, I tried to rebase off [Will's refactor](https://github.com/willbeason/frameworks/commit/7078cd84b1f2f4f99a5b7acc79e000f88d26f214) which fully moves `Handler` into it's own package, but quickly regretted it. Without Will's client test changes, there will already exist a `handler` type in client. We would have to make a bunch of named imports, only to clean them up later once Will's client changes are merged in.

I think the easiest solution for now is to copy the `TargetHandler` file into the new `handler` package, and defer the proper refactor until Will's PR

@willbeason does this sound reasonable?

**Review by willbeason — APPROVED** [review]

File: constraint/pkg/client/client.go [hunk 4]
```diff
@@ -5,12 +5,13 @@ import (
...
 	"errors"
 	"fmt"
-	clienterrors "github.com/open-policy-agent/frameworks/constraint/pkg/client/errors"
 	"path"
 	"sort"
 	"strings"
 	"sync"
 
+	clienterrors "github.com/open-policy-agent/frameworks/constraint/pkg/client/errors"
+
```
Comment (willbeason) [inline constraint/pkg/client/client.go:14]: remove newline after `clienterrors` import. Ditto others.

File: constraint/pkg/client/drivers/local/local_unit_test.go [hunk 5]
```diff
@@ -3,10 +3,11 @@ package local
 import (
 	"context"
 	"errors"
-	errors2 "github.com/open-policy-agent/frameworks/constraint/pkg/client/errors"
 	"sort"
 	"testing"
 
+	errors2 "github.com/open-policy-agent/frameworks/constraint/pkg/client/errors"
```
Comment (willbeason) [inline constraint/pkg/client/drivers/local/local_unit_test.go]: Rename "errors2" to "clienterrors" for consistency

File: constraint/pkg/client/crds/validate.go [hunk 6]
```diff
@@ -4,8 +4,9 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/open-policy-agent/frameworks/constraint/pkg/client/errors"
```
Comment (willbeason) [inline constraint/pkg/client/crds/validate.go:7]: Rename to `clienterrors` for consistency.

**Review by willbeason — APPROVED** [review]
Just minor changes, otherwise LGTM

File: constraint/pkg/client/client.go (hunk 4 above)
Comment (davis-haba) [reply constraint/pkg/client/client.go]: done

File: constraint/pkg/client/crds/validate.go (hunk 6 above)
Comment (davis-haba) [reply constraint/pkg/client/crds/validate.go]: done

**Review by maxsmythe — APPROVED** [review]
LGTM after lint error fixes

**Comment by willbeason** [comment]
You should be able to fix the lint errors with `golangci-lint run --fix`

**Review by ritazh — APPROVED** [review]
lgtm

**Pull request merged**